macros/src/lib.rs

```diff
@@ -131,10 +89,151 @@ impl DerivedTS {
             #export
         }
     }
+
+    /// Returns an expression which evaluates to the TypeScript name of the type, including generic
+    /// parameters.
+    fn name_with_generics(&self) -> TokenStream {
+        let name = &self.ts_name;
+        let mut generics_ts_names = self
+            .generics
+            .params
+            .iter()
+            .filter_map(|g| match g {
+                GenericParam::Lifetime(_) => None,
+                GenericParam::Type(ty) => Some(&ty.ident),
+                GenericParam::Const(_) => None,
+            })
+            .map(|generic| quote!(<#generic as ts_rs::TS>::name()))
+            .peekable();
+
+        if generics_ts_names.peek().is_some() {
+            quote! {
+                format!("{}<{}>", #name, vec![#(#generics_ts_names),*].join(", "))
+            }
+        } else {
+            quote!(#name.to_owned())
+        }
+    }
+
+    /// Generate a dummy unit struct for every generic type parameter of this type.
+    /// Example:
+    /// ```ignore
+    /// struct Generic<A, B, const C: usize> { /* ... */ }
+    /// ```
+    /// has two generic type parameters, `A` and `B`. This function will therefor generate
+    /// ```ignore
+    /// struct A;
+    /// impl ts_rs::TS for A { /* .. */ }
+    ///
+    /// struct B;
+    /// impl ts_rs::TS for B { /* .. */ }
+    /// ```
+    fn generate_generic_types(&self) -> TokenStream {
+        let generics = self.generics.params.iter().filter_map(|g| match g {
+            GenericParam::Lifetime(_) => None,
+            GenericParam::Type(t) => Some(t.ident.clone()),
+            GenericParam::Const(_) => None,
+        });
+
+        quote! {
+            #(
+                #[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, Ord, PartialOrd)]
+                struct #generics;
+                impl std::fmt::Display for #generics {
+                    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                        write!(f, "{:?}", self)
+                    }
+                }
+                impl TS for #generics {
+                    fn name() -> String { stringify!(#generics).to_owned() }
+                    fn transparent() -> bool { false }
+                }
+            )*
+        }
+    }
```

Comment:
Here, these dummy types are generated, named the same as the generic parameters.  
I thought about only defining one dummy type, so e.g `#[docs(hidden)] pub struct Dummy<const NAME: &'static str>`.  
However, const-generics don't work with strings yet. Maybe something for the future to improve compile times.